include/boost/url/pct_encoded_view.hpp

```diff
@@ -0,0 +1,696 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_PCT_ENCODED_VIEW_HPP
+#define BOOST_URL_PCT_ENCODED_VIEW_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/pct_encoding.hpp>
+#include <type_traits>
+#include <iterator>
+#include <iosfwd>
+
+namespace boost {
+namespace urls {
+
+/** A view of the percent decoded characters in a string
+
+    This class is a lightweight wrapper that presents
+    a view of an underlying sequence of percent-encoded
+    characters mutating or copying it.
+
+    The reference objects do not own the underlying
+    character buffer, are inexpensive to create and
+    copy. This avoid any unnecessary memory allocations
+    when working with decoded strings, and transfers
```

Comment:
You can say "The caller must opt-in to perform expensive, possibly-allocating decoding operations which produce a contiguous character buffer by calling @ref append_to, @ref assign_to,  or @ref to_string."